hetmech/hetmat.py

```diff
@@ -197,6 +224,71 @@ def permutations(self):
             permutations[name] = permutation
         return permutations
 
+    def permute_graph(self, num_new_permutations=None, namer=None, start_from=None,
+                      multiplier=10, excluded_pair_set=set(), seed=0, log=True):
+        """
+        Generate and save permutations of the HetMat adjacency matrices.
+
+        Parameters
+        ----------
+        num_new_permutations : int
+            The number of new, permuted HetMats to generate
+        namer : generator
+            Yields the names of new permutations. Cannot pass names of existing permutations
+        start_from : str
+            Name of permutation to use as starting point. For multiple permutations,
+            the first permutation starts from start_from, and future permutations
+            continue from the previous one.
+        multiplier : int
+            How many attempts to make when cross-swapping edges.
+        excluded_pair_set : set
+            Pairs of nodes which are not to be changed
+        seed : int
+            Random seed for generating new permutations
+        log : bool
+            When log=True, give information about permutation attempts
+        """
+        if namer is None:
+            # If no namer given, continue increasing names by one for new permutations
+            def namer():
+                i = 1
+                while True:
+                    yield str(i).zfill(3)
+                    i += 1
+            namer = namer()
+
+        logging_stats = list()
+        for _ in range(num_new_permutations):
+            permutation_name = next(namer)
+            name = f'{permutation_name}.hetmat/'
+            new_permutation_path = self.permutations_directory.joinpath(name)
+
+            new_hetmat = HetMat(new_permutation_path, initialize=True)
+            new_hetmat.is_permutation = True
+            new_hetmat.metagraph_path.symlink_to('../../metagraph.json')
+            new_hetmat.nodes_directory.rmdir()
+            new_hetmat.nodes_directory.symlink_to('../../nodes', target_is_directory=True)
+
+            if start_from is not None:
```

Comment:
Would be possible to simplify by renaming `original_hetmat` to `start_from`. User could then pass `None` for self, a hetmat for a specific loaded hetmat, or a str for the name of a permtuation.